### ISSUE TITLE
Setting title of email to non-standard entry deletes the email.

### ISSUE THREAD

**Issue by NukeManDan** [nextcloud/contacts#1705]
### Describe the bug
I attempted to set an email field name (clicking to the left of the email) to something, previously imported with no field name. On setting this, the email field is lost completely. No errors.

**To Reproduce**
1. go to a contact
2. click on area to left of email field ot set the title of that email address field
3. press `enter`
4. email is gone

**Expected behavior**
set the name, keep the email.

**Actual behavior**
email is gone without ability to recover.

**Screenshots**
![image](https://user-images.githubusercontent.com/35669742/87588285-9d0a8200-c6a0-11ea-95cf-8773ff784eec.png)

clicked here, set the field (cannot screenshot) then email is lost

### Server configuration

**Operating system**: linux - ubuntu 20 sever

**Nextcloud version:** 19 -- using the VM build in this org's repo list

**Contacts version:** most recent as of today

**Comment by NukeManDan** [comment]
NOTE another issue that we should log - the field to name the email address is invisible (at least in dark mode, light not tested)

**Comment by NukeManDan** [comment]
Also of note: it seems that a capital letter starting the title of the email may be the cause! I tried "Old Gmail" and this was rejected without error. "old gmail" worked.

The behaviour is erroneous though, sometimes it sets, some times it does not. Sometimes it give the specified name in the drop-down for further entries, sometimes it does not. :man_shrugging: 

This may be ralted to the use of `" "` the space in the field?

**Comment by skjnldsv** [comment]
Right, this is to search amongst existing types, not to enter a custom one.
